pulsar/consumer_partition.go

```diff
@@ -247,7 +294,7 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 	if msgMeta.NumMessagesInBatch != nil {
 		numMsgs = int(msgMeta.GetNumMessagesInBatch())
 	}
-	messages := make([]*message, numMsgs)
+	messages := make([]*message, 0)
```

Comment:
we can do this `make([]*message, 0, numMsgs)` then append will work and we don't ever have to reallocate the slice.

Reply:
The problem here is that we are doing the filtering, so the actual number of messages might be lower than that